lang/LangSource/SC_CLIOptions.cpp

```diff
@@ -0,0 +1,284 @@
+        )
+        (
+            "ide,i",
+            po::value<string>()->notifier([&](string ide_name) {
+                SC_Filesystem::instance().setIdeName(ide_name);
+            }),
+            "Set IDE name"
+        )
+        (
+            "standalone,a",
+            po::bool_switch(&terminalOptions.mStandalone)->default_value(terminalOptions.mStandalone),
+            "Standalone mode (exclude SCClassLibrary and user and system Extensions folders from search path)"
+        )
+    ;
+    // clang-format on
+
+    return description;
+}
+
+po::options_description CLIOptions::buildLanguageConfigDescription() {
+    po::options_description description("Language config options (overwrites provided yaml entries)");
+
+    // clang-format off
+    description.add_options()
+        (
+            "include-path",
+            po::value<vector<string>>(),
+            "Class library path to be included for searching (allows multiple mentions)"
+        )
+        (
+            "exclude-path",
+            po::value<vector<string>>(),
+            "Class library path to be to excluded from searching (overrides includePaths, allows multiple mentions)"
+        )
+        (
+            "post-inline-warnings",
+            po::bool_switch()->default_value(false),
+            "Boolean flag to turn on warnings about missing inline opportunities."
+        )
+        // exclude-default-paths is not included as this behavior is set via the `--standalone` flag
+    ;
+    // clang-format on
+    return description;
+}
+
+void CLIOptions::parseLanguageConfig(po::variables_map& vm) {
+    // see HelpSource/Classes/LanguageConfig.schelp
+    if (vm.at("post-inline-warnings").as<bool>()) {
+        gLanguageConfig->setPostInlineWarnings(true);
+    };
+
+    if (vm.count("include-path")) {
+        for (auto includePath : vm["include-path"].as<vector<string>>()) {
+            gLanguageConfig->addIncludedDirectory(SC_Codecvt::utf8_str_to_path(includePath));
+        }
+    }
+    if (vm.count("exclude-path")) {
+        for (auto excludePath : vm["exclude-path"].as<vector<string>>()) {
+            gLanguageConfig->addExcludedDirectory(SC_Codecvt::utf8_str_to_path(excludePath));
+        }
+    }
+}
+
+// converts e.g. 1k to 1024
+int CLIOptions::parseMemoryString(std::string input) {
+    if (input.empty()) {
+        return 0;
+    }
+
+    char suffix = input.back();
+    int multiplier = 1;
+
+    if (suffix == 'k' || suffix == 'K') {
+        multiplier = 1024;
+    } else if (suffix == 'm' || suffix == 'M') {
+        multiplier = 1024 * 1024;
+    } else if (suffix == 'g' || suffix == 'G') {
+        multiplier = 1024 * 1024 * 1024;
+    } else if (!isdigit(suffix)) {
+        cerr << "Unknown modifier '" << suffix << "'" << endl;
+        return 0;
+    }
+
+    std::string numberPart = std::isdigit(suffix) ? input : input.substr(0, input.size() - 1);
+    int value = boost::lexical_cast<int>(numberPart);
```

Comment:
Note: `boost::lexical_cast` throws an `boost::bad_lexical_cast` exception if the conversion is not successfully. I guess we should catch it here, print an error and return 0?